File: Algo/hr_implement/3/appendAndDelete.js
function appendAndDelete(s, t, k) {
    if (s.length + t.length < k) {
      return 'Yes';
    }
    let diff = 0;
    for (let i in s) {
      if (i >= t.length) {
        diff = i;
        break;
      } else if (s[i] != t[i]) {
        diff = i;
        break;
      }
    }
    diff = s.length - diff + (t.length - diff);
  
    if (diff <= k && (diff - k) % 2 == 0) {
      return 'Yes';
    } else {
      return 'No';
    }
  }
  
  let s = 'hackerhappy',
    t = 'hackerrank',
    k = 9;
  console.log(appendAndDelete(s, t, k));
  

File: Algo/hr_implement/3/appendAndDelete.py
"""
Append, Delete to match two strings in given moves
"""

def appendAndDelete(s, t, k):
	if (len(s)+ len(t)) < k:
		return 'Yes'

	i = 0
	for a in s:
		if i >= len(t) or a != t[i]:
			break
		i +=1

	diff = (len(s) - i) + (len(t) - i)
	print i, diff
	if (diff<=k and ((diff-k) % 2 ==0)):
		return 'Yes'
	else:
		return 'No'

if __name__ == "__main__":
	s = 'hackerhappy' 
	t = 'hackerrank'
	k = 9
	"""s = 'ashley'
	t = 'ash'
	k = 2"""
	"""s = 'aba'
	t = 'aba'
	k = 7"""
	print appendAndDelete(s,t,k)
